src/xml/reader.cr

```diff
@@ -31,11 +31,7 @@ class XML::Reader
   # See `XML::ParserOptions.default` for default options.
   def initialize(str : String, options : XML::ParserOptions = XML::ParserOptions.default)
     @reader = LibXML.xmlReaderForMemory(str, str.bytesize, nil, nil, options)
-    LibXML.xmlTextReaderSetErrorHandler @reader, ->(arg, msg, severity, locator) do
-      msg_str = String.new(msg).chomp
-      line_number = LibXML.xmlTextReaderLocatorLineNumber(locator)
-      raise Error.new(msg_str, line_number)
-    end
+    LibXML.xmlTextReaderSetStructuredErrorHandler(@reader, ->Error.structured_callback, Box.box(@errors))
```

Comment:
I notice this is a breaking change... but I don't understand why "read from string" should raise, while "read from io" should collect errors instead :raised_eyebrow: